python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -132,6 +139,12 @@ export default function aiconfigReducer(
     case "ADD_PROMPT_AT_INDEX": {
       return reduceInsertPromptAtIndex(state, action.index, action.prompt);
     }
+    case "DELETE_PROMPT": {
+      return {
+        ...state,
+        prompts: state.prompts.filter((prompt) => prompt._ui.id !== action.id),
```

Comment:
Why do we need to nest this extra `_ui.id` field instead of using `id` directly in https://github.com/lastmile-ai/aiconfig/blob/ca59d2c1657fe6e2655f1a0c25180f9407b97cab/python/src/aiconfig/editor/client/src/shared/types.ts#L13-L15

Reply:
I think it makes it really clear that there's a UI-only distinction between ClientPrompt and Prompt and makes it clearer to reason about for conversion between the two where needed